src/components/chat/components/list/index.ts

```diff
@@ -0,0 +1,27 @@
+import Component from "../../../../utils/component";
+import { useEventBus, uid } from "../../../../utils";
+import Item from "../item";
+import template from "./index.tem";
+
+const [on] = useEventBus;
+export default class List extends Component {
+  constructor(props: P) {
+    super({ ...props, template, "Chat.Item": Item });
+  }
+
+  render() {
+    on("ChatItemSelected", (chat: any) => {
+      const items = this.element.querySelectorAll(".chat-item");
+      [...items].forEach((item) =>
+        item.setAttribute("data-active", String(Number(item.getAttribute("chat-id")) === chat.id))
+      );
+    });
+
+    const { chats } = this.props;
+    //prettier-ignore
+    const list = chats? chats.map((chat: any) => new Item({ chat, className: `chat__item` })) : "";
```

Comment:
сейчас list может быть либо массивом, либо строкой. Это не совсем корректно, потому что мы используем typescript и когда переменная может быть разного типа, это создает неоднозначность, что всегда плохо. Код должен быть максимально понятен и предсказуем. Я предлагаю написать здесь так:
```suggestion
    const { chats = [] } = this.props;
    //prettier-ignore
    const list = chats.map((chat: any) => new Item({ chat, className: `chat__item` }));
```

тогда list в любом случае будем массивом - либо пустым, либо с элементами. Нам не нужно будет запоминать что в нем лежит - строка или массив